lib/message_bus.rb

```diff
@@ -599,6 +599,24 @@ def client_message_filters
     @config[:client_message_filters]
   end
 
+  # @yieldparam [String] metric_name some name for the metric being recorded
+  def tracer(&blk)
+    @config[:tracer] = blk
+  end
+
+  # Executes the provided block of code within the context of a probe
+  # Intended to be managed by a consuming application which will define
+  # the `:tracer` config option.
+  #
+  # @param [String] metric_name some name for the metric being recorded
+  def trace(metric_name, &blk)
+    if @config[:tracer]
```

Comment:
A bit safer but very minor.

```suggestion
    if @config[:tracer].is_a?(Proc)
```

Reply:
The precedent in other similar cases in message_bus is to assume it is callable if set.